packages/pillarbox-playlist/src/pillarbox-playlist-modal.js

```diff
@@ -0,0 +1,228 @@
+ * It is designed to manage and display a playlist with various controls.
+ */
+class PlaylistMenuDialog extends ModalDialog {
+  /**
+   * Handles the 'statechanged' event when triggered. This method serves as a proxy to
+   * the main `statechanged` handler, ensuring that additional logic can be executed or
+   * making it easier to detach the event listener later.
+   *
+   * @private
+   */
+  _onPlaylistStateChanged = ({ changes }) => {
+    if ('items' in changes) {
+      this.removeItems();
+      this.renderItems();
+    }
+
+    if ('currentIndex' in changes) {
+      this.select(changes.currentIndex.to);
+    }
+  };
+
+  /**
+   * Creates an instance of PlaylistMenuDialog.
+   *
+   * @param {import('@srgssr/pillarbox-web').Player} player - The pillarbox player instance.
+   * @param {Object} options - Options for the dialog.
+   */
+  constructor(player, options) {
+    options.temporary = false;
+
+    super(player, options);
+    this.fill();
+    this.renderComponent();
+    this.playlist().on('statechanged', this._onPlaylistStateChanged);
+  }
+
+  buildCSSClass() {
+    return `pbw-playlist-dialog ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * Dispose of the PlaylistMenuDialog instance.
+   */
+  dispose() {
+    this.playlist().off('statechanged', this._onPlaylistStateChanged);
+    super.dispose();
+  }
+
+  /**
+   * Get the playlist instance associated with the player.
+   *
+   * @returns {import('pillarbox-playlist.js').default} The playlist instance.
+   */
+  playlist() {
+    return this.player().pillarboxPlaylist();
+  }
+
+  /**
+   * Update the playlist item UI with the selected index.
+   *
+   * @param {number} index - The index of the item to select.
+   */
+  select(index) {
+    this.children()
+      .filter(child => child.name() === 'PillarboxPlaylistMenuItem')
+      .forEach(child => child.selected(index === child.options().index));
+  }
+
+  /**
+   * Remove all playlist items from the dialog.
+   */
+  removeItems() {
+    this.children()
+      .filter(child => child.name() === 'PillarboxPlaylistMenuItem')
+      .forEach(child => this.removeChild(child));
+  }
+
+  /**
+   * Render the playlist items in the dialog.
+   */
+  renderItems() {
+    this.playlist().items.forEach((item, index) => this.addChild(
+      new PillarboxPlaylistMenuItem(this.player(), { item, index, name: 'PillarboxPlaylistMenuItem' })
+    ));
+  }
+
+  /**
+   * Create the playlist control buttons.
+   *
+   * @returns {Component} The component containing the playlist control buttons.
+   */
+  createControls() {
+    const playlistControls = new Component(
+      this.player(),
+      { name: 'PlaylistControls', className: 'pbw-playlist-controls' }
+    );
+
+    playlistControls.addChild(this.createPreviousItemButton());
+    playlistControls.addChild(this.createRepeatButton());
+    playlistControls.addChild(this.createSuffleButton());
+    playlistControls.addChild(this.createNextItemButton());
```

Comment:
In terms of visual display, having the previous/next buttons at either end is more visually pleasing. However, for the sake of accessibility and display logic, I would have tended to put these two buttons together.

![Screenshot from 2024-05-24 16-14-51](https://github.com/SRGSSR/pillarbox-web-suite/assets/34163393/321c1a05-e090-46cd-87ef-70baa7135d99)

or something else.

Reply:
I've grouped them like this so that there 2 distinctive groups with similar funcitonalities:

<img width="462" alt="image" src="https://github.com/SRGSSR/pillarbox-web-suite/assets/11271371/98407110-5a51-4f76-a6e4-b361aa3e4d1b">